### DISCUSSION TITLE
How do I check for clicking the console window close button?

### DISCUSSION THREAD

**Discussion by reapercreap** [pyinstaller/pyinstaller#6481] [Help] [answered]
I think the question is clear.
`import logging
try:
    logging.basicConfig(filename=".\\test.log", level=logging.ERROR)
    input("waiting for input")
except BaseException as e:
    logging.error(e)
    exit(1)`

Thank you, would be really nice I need it for work.

**Comment by bwoodsend** [comment, marked as answer]
If by *check for* you mean detect then it's the same as with normal Python - you can't. If you're trying to add on-exit behaviour then use [axtexit](https://docs.python.org/3/library/atexit.html). Otherwise, I can't tell what you're trying to do.

**Reply by reapercreap** [reply]
So first thanks for the reply. I have ignored the problem as is seemed to be not necessary but now I wanted to try it out and after programming with win32 api the last days I think I learned what you mean. The created pyinstaller xxxx.exe is just a wrapper that adds a temporary MEIPASS environment variable and created a directory where it extracts the carried Python environment and creating a new process with the python interpreter starting the script, so pythons, doesn't know anything about the program from which it has been started? It is like a little executable that can start other programs. Am I right with this?
Thank you.

**Reply by bwoodsend** [reply]
No program, irregardless of whether it uses PyInstaller or even Python, knows anything about its parent process. If you want that information then you have to ask the OS for it via its native APIs or there's a cross platform wrapper library called [psutil](https://github.com/giampaolo/psutil) which you could use.
```python
>>> psutil.Process().parents()
[psutil.Process(pid=692031, name='fish', status='sleeping', started='10:29:36'), psutil.Process(pid=1433, name='xfce4-terminal', status='sleeping', started='2022-01-28 18:07:01'), psutil.Process(pid=1, name='systemd', status='sleeping', started='2022-01-28 18:06:48')]
```
(Here `fish` is the name of my terminal shell, `xfce4-terminal` is the terminal window and `systemd` is some kind of top level Linux process manager.)
